crates/runtime/src/accelerated_table.rs

```diff
@@ -80,6 +81,11 @@ impl AcceleratedTable {
         refresh_mode: RefreshMode,
         refresh_interval: Option<Duration>,
         refresh_sql: Option<String>,
+        time_column: Option<String>,
+        time_format: Option<TimeFormat>,
+        retention_check_interval: Option<Duration>,
+        retention_period: Option<Duration>,
+        retention_enabled: bool,
```

Comment:
Not a fan of all these parameters - can we leverage the type system more effectively here? Maybe something similar to what Sergei did for the refresh mode enum to only pass the data relevant to a particular case.

Reply:
All of these new parameters are only related to retention. Let's wrap them with a struct in an Option<> and then you can also get rid of `retention_enabled`

Reply:
yeah actually about to move it to use enum.